src/webob/acceptparse.py

```diff
@@ -268,6 +667,1250 @@ def _match(self, mask, item):
+           :meth:`AcceptLanguageValidHeader.quality`).
+
+        :param offer: (``str``) language tag offer
+        :return: (``float``) ``1.0``.
+
+        When the ``Accept-Language`` header is invalid or not in the request,
+        all offers are equally acceptable, so 1.0 is always returned.
+        """
+        warnings.warn(
+            'The behavior of .quality for the AcceptLanguage classes is '
+            'currently being maintained for backward compatibility, but the '
+            'method may be deprecated in the future, as its behavior is not '
+            'specified in (and currently does not conform to) RFC 7231.',
+            PendingDeprecationWarning,
+        )
+        return 1.0
+
+
+class AcceptLanguageNoHeader(_AcceptLanguageInvalidOrNoHeader):
+    """
+    Represent when there is no ``Accept-Language`` header in the request.
+
+    This object should not be modified. To add to the header, we can use the
+    addition operators (``+`` and ``+=``), which return a new object (see the
+    docstring for :meth:`AcceptLanguageNoHeader.__add__`).
+    """
+
+    def __init__(self):
+        """
+        Create an :class:`AcceptLanguageNoHeader` instance.
+        """
+        self._header_value = None
+        self._parsed = None
+        self._parsed_nonzero = None
+
+    @property
+    def header_value(self):
+        """
+        (``str``) The header value.
+
+        As there is no header in the request, this is ``None``.
+        """
+        return self._header_value
+
+    @property
+    def parsed(self):
+        """
+        (``list``) Parsed form of the header.
+
+        As there is no header in the request, this is ``None``.
+        """
+        return self._parsed
+
+    def __add__(self, other):
+        """
+        Add to header, creating a new header object.
+
+        `other` can be:
+
+        * a ``str``
+        * a ``dict``, with language ranges as keys and qvalues as values
+        * a ``tuple`` or ``list``, of language range ``str``s or of ``tuple``
+          or ``list`` (language range, qvalue) pairs (``str``s and pairs can be
+          mixed within the ``tuple`` or ``list``)
+        * any object that returns a value for `__str__`
+        * an :class:`AcceptLanguageValidHeader`,
+          :class:`AcceptLanguageNoHeader`, or
+          :class:`AcceptLanguageInvalidHeader` instance
+
+        If neither operands are empty (header value ``''``) or
+        ``None``/:class:`AcceptLanguageNoHeader`, the two header values are
+        joined with ``', '``.
+        """
+        if isinstance(other, AcceptLanguageNoHeader):
+            return self.__class__()
+
+        if isinstance(other, AcceptLanguageValidHeader):
+            return create_accept_language_header(
+                header_value=other.header_value
+            )
+
+        if isinstance(other, AcceptLanguageInvalidHeader):
+            return AcceptLanguageInvalidHeader(header_value=other.header_value)
```

Comment:
I've made the changes: `InvalidHeader`s and invalid header values are now treated like `NoHeader`s when `__add__`- and `__radd__`ing.